overlord/snapstate/refreshhints.go

```diff
@@ -121,3 +136,51 @@ func (r *refreshHints) Ensure() error {
 	}
 	return r.refresh()
 }
+
+func refreshHintsFromCandidates(st *state.State, updates []*snap.Info, ignoreValidationByInstanceName map[string]bool, deviceCtx DeviceContext) ([]*refreshCandidate, error) {
+	if ValidateRefreshes != nil && len(updates) != 0 {
+		userID := 0
+		var err error
+		updates, err = ValidateRefreshes(st, updates, ignoreValidationByInstanceName, userID, deviceCtx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	hints := []*refreshCandidate{}
+	for _, update := range updates {
+		var snapst SnapState
+		if err := Get(st, update.InstanceName(), &snapst); err != nil {
+			return nil, err
+		}
+
+		flags := snapst.Flags
+		flags.IsAutoRefresh = true
+		flags, err := earlyChecks(st, &snapst, update, flags)
```

Comment:
(nitpick) I found this a big confusing to read, maybe something like:
```
 		flags, err := earlyChecks(st, &snapst, update, &snapst.Flags{IsAutoRefresh: true})
```
to make it more clear what is going on? But maybe just me again :)

Reply:
We're using snapst.Flags here, not snapstate.Flags{}.

Reply:
Indeed, sorry, was confused. This is fine